index.ts

```diff
@@ -302,15 +320,26 @@ function transpile(
 
 		sourceFile.getFunctions().forEach(generateFunctionDocumentation);
 
-		const result = project.emitToMemory()?.getFiles()?.[0]?.text;
+		let result = project.emitToMemory()?.getFiles()?.[0]?.text;
 		if (result) {
+			if (!result.startsWith(protectCommentsHeader)) {
+				throw Error(
+					`internal error: protectCommentsHeader is missing in output\n\n` +
+					`protectCommentsHeader: ${JSON.stringify(protectCommentsHeader)}\n` +
+					`result: ${JSON.stringify(result.slice(protectCommentsHeader.length + 100) + " ...")}`
```

Comment:
helpful for debugging, when ts-morph/typescript returns an unexpected value
we want to remove exactly what we added

Reply:
Sorry, I was referring to `JSON.stringify(protectCommentsHeader)`. Since it's a defined constant variable where we know its content, in what scenario would the `JSON.stringify` call produce a more desirable output than just including the variable without the wrapping function call?

Reply:
exact diff